### PR TITLE
Task3

### PR DESCRIPTION
**Pull request by TatsianaSlapik**
Description
Install Storybook

Install and configure [Storybook](https://storybook.js.org/) by following guide from the official website. In the end you should have a "storybook" npm script. When running "npm run storybook" it should start Storybook and open it in your browser.

 

Write Storybook stories for existing components

Create a story file for every component you created in previous modules. You should have at least one story for each of the following components (in total minimum 3):

⚛️ Counter
⚛️ SearchForm
⚛️ GenreSelect
You can write more stories to cover several states of GenreSelect or SearchForm, but this is optional.

 

Implement other components

Based on [Figma prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/CDP-Home-Task-%E2%80%93-React-v1?node-id=0%3A2) implement the following components:

⚛️ MovieTile

This component should render a movie tile from the list of movies.

It should take properties to receive image url, movie name, release year, and a list of relevant genres. Alternatively, you can define one component property to take an object with all movie info. Additionally, the component should receive a callback property to capture click event.

Optionally, you can implement a context menu popup that opens when a user clicks on three dots button and contains "Edit" and "Delete" menu items.

⚛️ MovieDetails

This component will render movie details when a movie is selected from the list (clicked). The details include movie poster image on the left and the rest of info on the right.

The component should take properties to receive image url, movie name, release year, rating, duration and a description. Alternatively, you can specify a single property that accepts an object with all movie info.

⚛️ SortControl

This component will render a label "Sort by" and a select control to the right. Select should have the following options:

Release Date
Title
The component should take a property that specifies current selection. Additionally, it should take a callback property to handle selection changes. The callback should be called every time a user changes "Sort by" value. The new value should be passed in callback arguments.

Use Storybook when implementing every individual component. Write a story for every component you create. It will help you to check the result and play around with some interactivity before you embed components into the app.

Although, it's not required to implement high-fidelity design as per design prototype, it still makes sense to apply some styling to your components, so that you practice in styling React components and your final result looks good.

 

Write tests

Cover new components with tests using jest and @testing-library/react. Verify necessary data is rendered as well as that behavior works correctly. You can write snapshot tests to cover the rendering. Although, remember that snapshot tests are fragile and will fail any time you change your component markup.

 

Evaluation criteria:

Turn back for updates - Storybook is not installed or doesn't open.
Any of required components is not implemented.
Tests are absent or they don't pass.
3 - Storybook is installed and opens.
All new components from task description are implemented and work without major issues. Some minor issues are acceptable.
New components are not visible in Storybook.
Some tests are missing or they fail when executed.
4 - Storybook is installed and opens.
All new components from task description are implemented and work without major issues. Some minor issues are acceptable.
New components are visible in Storybook.
Some tests are missing or they fail when executed.
5 - Storybook is installed and opens.
All new components from task description are implemented as described, including optional behavior. Alternatively, additional components from design prototype are implemented (not included in task description).
All components from task description have stories in Storybook and are visible when Storybook is opened.
Tests are written for all components and they all pass. Edge cases are covered.

### CODE REVIEW

File: react_app/src/components/Input/Input.stories.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const InputStories = () => (
+  <Input
+    name='add text'
+    title='Please add text'
+    type='text'
+    value='add text'
+    onChange={() => {}}
+    defaultValue='add text'
+  />
+);
```
Comment (aliskar) [inline react_app/src/components/Input/Input.stories.js:18]: You can add `Controls` into storybook UI by doing this:
```jsx
const Template = (args) => <Input {...args} />;

export const Default = Template.bind({});

Default.args = {
  name:'add text',
  title:'Please add text',
  type:'text',
  value:'add text',
  onChange:() => {},
  defaultValue:'add text',
};
```
<details>
<summary>Storybook UI</summary>

image
</details>

File: react_app/src/data/data.js [hunk 2]
```diff
@@ -1,7 +1,79 @@
...
+
 export const genres = [
   { id: 0, genre: "All" },
   { id: 1, genre: "Documentary" },
   { id: 2, genre: "Comedy" },
   { id: 3, genre: "Horror" },
   { id: 4, genre: "Crime" },
 ];
+
+export const FILMS_DATE = [
```
Comment (aliskar) [inline react_app/src/data/data.js:16]:
```suggestion
export const FILMS_DATA = [
```

File: react_app/src/components/MovieTile/MovieTile.css [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    height: 60px;
+    line-height: 40px;
+
+    text-align: center;
+    border: 1px solid #979797;
+    border-radius: 4px;
+    margin-left: 30px;
+
+    width: 66px;
+    height: 26px;
```
Comment (aliskar) [inline react_app/src/components/MovieTile/MovieTile.css:47]: `width` and `height` used twice

File: react_app/src/components/SortControl/SortControl.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    padding: 18px;
+    margin-left: 215px;
+}
+.sortBy {
+    font-family: "Montserrat";
+    font-style: normal;
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 20px;
+    letter-spacing: 0.888889px;
```
Comment (aliskar) [inline react_app/src/components/SortControl/SortControl.css:13]: wow, that's precision! 🔍😁

**Review by aliskar — APPROVED** [review]
well done
